aioredlock/algorithm.py

```diff
@@ -67,20 +54,46 @@ def _validate_retry_delay(self, attribute, value):
     def log(self):
         return logging.getLogger(__name__)
 
-    async def lock(self, resource):
+    async def _auto_extend(self, lock):
+        """
+        Tries to reset the lock's lifetime to lock_timeout every 0.6*lock_timeout automatically
+        In case of fault the LockError exception will be raised
+        :param lock: :class:`aioredlock.Lock`
+        :raises: LockError in case of fault
+        """
+
+        await asyncio.sleep(0.6 * self.internal_lock_timeout)
+        try:
+            await self.extend(lock)
+        except Exception:
+            self.log.debug('Error in extending the lock "%s"',
+                           lock.resource)
+
+        self._watchdogs[lock.resource] = asyncio.ensure_future(self._auto_extend(lock))
```

Comment:
This recursion could cause stack overflow in case of the long living lock.

Reply:
why? Actually there's always at most one auto_extend task pending in event_loop for one lock, so there wont be stack overflow

Reply:
Yup... I'm wrong with that.
Sorry.